Reject queries with unknown query params (WIP) (#869)
cloudevents/spec: discovery.md

format MUST be supported.  Likewise requests with `Accept` header of
`application/json` MUST be supported.

> Unkonwn query parameters on the APIs MUST be ignored.

Reviewer: ```suggestion
Unknown query parameters on the APIs MUST be ignored.
```

(If I can't make substantive contributions, I can at least help with typos...)

latest version of all Services available via this Discovery Enpoint.

The collection of services MAY be filtered by supplying one or more
[service attributes](#service-attributes) under the `filter` query parameter.
The general format for the `filter` query parameter is:

```

> ?filter=ATTRIBUTE=VALUE[,ATTRIBUTE=VALUE]

Author: Open question: how do people specify a comma in a value? Should we support an escape, or should we not allow multiple per "filter" and require multiple "filter" query params instead? I'm leaning toward forcing multiple query parameters.

latest version of all Services available via this Discovery Enpoint.

The collection of services MAY be filtered by supplying one or more
[service attributes](#service-attributes) under the `filter` query parameter.
The format for the `filter` query parameter MUST be:

```
?filter=ATTRIBUTE[=VALUE]
```

The `=VALUE` portion is OPTIONAL and if not present then the implied meaning
is that the specified attribute in the Service has a non-empty value.

Multiple attribute/value combinations MAY be specified as separate `filter`
query parameters. When there are multiple filters, the resulting set of
Services MUST only include ones that match all of the filters specified.
Matching of attribute names and values MUST be case sensitive.

Discovery endpoints MUST support filtering by the following attributes:

- `name`

> Requests with unsupported filter attributes, MUST reject the request with a

Reviewer: ```suggestion
Requests with unsupported filter attributes MUST be rejected with a
```

Or "Endpoints MUST reject requests with unsupported filter attributes, returning a `400 Bad Request` response."
